src/ApcuRateLimiter.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+declare(strict_types=1);
+
+namespace RateLimit;
+
+use RateLimit\Exception\CannotUseRateLimiter;
+use RateLimit\Exception\LimitExceeded;
+
+final class ApcuRateLimiter implements RateLimiter, SilentRateLimiter
+{
+    /** @var string */
+    private $keyPrefix;
+
+    public function __construct(string $keyPrefix = '')
+    {
+        if (!\extension_loaded('apcu') || \ini_get('apc.enabled') === '0') {
+            throw new CannotUseRateLimiter('APCu extension is not loaded or not enabled.');
+        }
+
+        if (\ini_get('apc.use_request_time') === '1') {
+            throw new CannotUseRateLimiter('APCu ini configuration "apc.use_request_time" should be set to "0".');
+        }
+
+        $this->keyPrefix = $keyPrefix;
+    }
+
+    public function limit(string $identifier, Rate $rate): void
+    {
+        $limitKey = $this->limitKey($identifier, $rate->getInterval());
+
+        $current = $this->getCurrent($limitKey);
+        if ($current >= $rate->getOperations()) {
+            throw LimitExceeded::for($identifier, $rate);
+        }
+
+        $this->updateCounter($limitKey, $rate->getInterval());
+    }
+
+    public function limitSilently(string $identifier, Rate $rate): Status
+    {
+        $interval = $rate->getInterval();
+        $limitKey = $this->limitKey($identifier, $interval);
+        $timeKey = $this->timeKey($identifier, $interval);
```

Comment:
Here's another idea for building a key with embedded time information:
```
$key = sprintf(
    '%s:%s',
    $this->keyPrefix.$identifier,
    floor(time() / $rate->getInterval())
);
```

The idea with second part is the put identifiers into consistent time buckets. Therefore `Status->resetAt` boils down to `strstr($key, ':') + $rate->getInterval()`, right?

Reply:
Although the idea of time buckets is good I'm afraid it's not feasible just like that, or I'm getting the idea wrong?

time() = 1590416425 (05/25/2020 @ 2:20pm (UTC))
for a 10 seconds interval, floor(1590416425 / 10) = 159041642
159041642 + 10 = 159041652 (01/15/1975 @ 6:14pm (UTC))

Reply:
You're right, I was just thinking out loud. These buckets are useful for auto-expiry, calculating TTL is possible, but it's not easy/accurate.